lib/rules/eqeqeq.js

```diff
@@ -79,11 +83,17 @@ module.exports = function(context) {
                 return;
             }
 
-            context.report(
-                node, getOperatorLocation(node),
-                "Expected '{{op}}=' and instead saw '{{op}}'.",
-                {op: node.operator}
-            );
+            context.report({
+                nodeType: node.type,
+                loc: getOperatorLocation(node),
+                message: "Expected '{{op}}=' and instead saw '{{op}}'.",
+                data: { op: node.operator },
+                fix: function(fixer) {
+                    return fixer.insertTextAfter(sourceCode.getTokenAfter(node.left).range, "=");
+
+                }
+            });
+
```

Comment:
Just to make sure I understood you correctly. You want two distinct signatures:
```js
context.report(node, message);
context.report({...});
```
Right?